clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10950,6 +10950,51 @@ def warn_imp_cast_drops_unaligned : Warning<
   InGroup<DiagGroup<"unaligned-qualifier-implicit-cast">>;
 
 // Function effects
+def warn_func_effect_violation : Warning<
+  "'%0' %select{function|constructor|destructor|lambda|block|constructor's member initializer}1 "
```

Comment:
The `%0` at the beginning is the name of the constraining attribute, so the diagnostic looks something like:

'nonblocking' lambda must not allocate or deallocate memory

Is that sufficient?

Reply:
Hmm, I DID miss that part, but I'm not sure how obvious it is why it is nonblocking.  It would be nice if we made it obvious it came from an attribute.

Reply:
How about

lambda with 'nonblocking' attribute must not allocate or deallocate memory

?

Reply:
either of those work, I think we're consistently inconsistent here which we go with.